go-controller/pkg/libovsdb/ops/router.go

```diff
@@ -775,32 +775,21 @@ func DeleteLogicalRouterStaticRoutesWithPredicate(nbClient libovsdbclient.Client
 }
 
 // DeleteLogicalRouterStaticRoutesWithPredicateOps looks up logical router static
-// routes from the cache based on a given predicate, and returns the ops to delete
-// them and remove them from the provided logical router
+// routes from the logical router of the specified name based on a given predicate,
+// and returns the ops to delete them and remove them from the provided logical router
 func DeleteLogicalRouterStaticRoutesWithPredicateOps(nbClient libovsdbclient.Client, ops []ovsdb.Operation, routerName string, p logicalRouterStaticRoutePredicate) ([]ovsdb.Operation, error) {
-	router := &nbdb.LogicalRouter{
-		Name: routerName,
+	deleted, err := GetRouterLogicalRouterStaticRoutesWithPredicate(nbClient, &nbdb.LogicalRouter{Name: routerName}, p)
```

Comment:
nit: I think it is better to find a different name that `deleted` here. It gives the impression the routes were already deleted and makes you look twice at the method that is being called here. I would just call it `routes` or `lrsrs`.